api/client/set.go

```diff
@@ -0,0 +1,104 @@
+package client
+
+import (
+	"fmt"
+	"strings"
+
+	Cli "github.com/docker/docker/cli"
+	flag "github.com/docker/docker/pkg/mflag"
+	"github.com/docker/docker/runconfig"
+	"github.com/docker/go-units"
+)
+
+// CmdSet sets resources of one or more containers.
+//
+// Usage: docker set [OPTIONS] CONTAINER [CONTAINER...]
+func (cli *DockerCli) CmdSet(args ...string) error {
+	cmd := Cli.Subcmd("set", []string{"CONTAINER [CONTAINER...]"}, Cli.DockerCommands["set"].Description, true)
+	flBlkioWeight := cmd.Uint16([]string{"-blkio-weight"}, 0, "Block IO (relative weight), between 10 and 1000")
+	flCPUPeriod := cmd.Int64([]string{"-cpu-period"}, 0, "Limit CPU CFS (Completely Fair Scheduler) period")
+	flCPUQuota := cmd.Int64([]string{"-cpu-quota"}, 0, "Limit CPU CFS (Completely Fair Scheduler) quota")
+	flCpusetCpus := cmd.String([]string{"-cpuset-cpus"}, "", "CPUs in which to allow execution (0-3, 0,1)")
+	flCpusetMems := cmd.String([]string{"-cpuset-mems"}, "", "MEMs in which to allow execution (0-3, 0,1)")
+	flCPUShares := cmd.Int64([]string{"#c", "-cpu-shares"}, 0, "CPU shares (relative weight)")
+	flMemoryString := cmd.String([]string{"m", "-memory"}, "", "Memory limit")
+	flMemoryReservation := cmd.String([]string{"-memory-reservation"}, "", "Memory soft limit")
+	flMemorySwap := cmd.String([]string{"-memory-swap"}, "", "Total memory (memory + swap), '-1' to disable swap")
+	flKernelMemory := cmd.String([]string{"-kernel-memory"}, "", "Kernel memory limit")
+
+	cmd.Require(flag.Min, 1)
+	cmd.ParseFlags(args, true)
+	if cmd.NFlag() == 0 {
+		return fmt.Errorf("You should always set one or more flags when use this command.")
```

Comment:
Minor thing but going back to a previous life where RFC2119 ruled everything, can you change this to something like:
`You must provide one or more flags when using this command` ?  The "should" doesn't mean its required, so "must" is more appropriate.